dkim_verify fails to find DKIM record, thinks it's just a TXT record.

**Describe the bug**
Upon receipt of an email sent from the same system, the dkim_verify plugin is failing to find the DKIM key in the DNS records. It reports that it is ignoring it as a TXT record.

**Expected behavior**
dkim_verify should understand the record.

**Observed behavior**
Here is the log output:

```
[DEBUG] [-] [dkim_verify] @dev.port87.com: DNS lookup oct2021._domainkey.dev.port87.com (timeout= 29s)
[DEBUG] [-] [dkim_verify] @dev.port87.com: ignoring TXT record: v=DKIM1;p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA10uf0NZ1Wn5GLWckB0TvONGAH8aZkFMErmwo9IS94ExrxqXpc8O6lUAbYRYSF2uRcLkPtlnK8lYdF3JjRbXt0RGx35UjF/5XWlS6103qG0ybCXO8kOiuoWszIzjKklMKiHq+D+E8S7F34NqRFD9nY5bsSDZbQrAq+Wdz/OkUs8qn29BFG/sfWF9whGWgfENtKp6uwBOQM,nCCOPS3dObcLw5OsNALmhsuy6Kx4PfkqDDD1u2K7qQm4XXrQsKo6n5O1sM8kLcO1ZmnjzllTb00O8tM7IhhUBv/KvMNye1dpnM8S7v7Mb9j0uXj9fNdAd4CpDVsvU1iLB6H3UT2JQRnPQIDAQAB                                                          
[DEBUG] [-] [dkim_verify] {"identity":"@dev.port87.com","selector":"oct2021","domain":"dev.port87.com","result":"invalid"}
[INFO] [54C26F57-A59F-49A4-BADA-054E6549D426.1] [dkim_verify] identity="@dev.port87.com" domain="dev.port87.com" selector="oct2021" result=invalid  (no key for signature)
[ERROR] [54C26F57-A59F-49A4-BADA-054E6549D426.1] [dkim_verify] dev.port87.com (no key for signature)
```

**Steps To Reproduce**
I'm not sure exactly how to reproduce this behavior. Here is a copy of my DKIM record from CloudFlare's DNS:

```
v=DKIM1;p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA10uf0NZ1Wn5GLWckB0TvONGAH8aZkFMErmwo9IS94ExrxqXpc8O6lUAbYRYSF2uRcLkPtlnK8lYdF3JjRbXt0RGx35UjF/5XWlS6103qG0ybCXO8kOiuoWszIzjKklMKiHq+D+E8S7F34NqRFD9nY5bsSDZbQrAq+Wdz/OkUs8qn29BFG/sfWF9whGWgfENtKp6uwBOQMnCCOPS3dObcLw5OsNALmhsuy6Kx4PfkqDDD1u2K7qQm4XXrQsKo6n5O1sM8kLcO1ZmnjzllTb00O8tM7IhhUBv/KvMNye1dpnM8S7v7Mb9j0uXj9fNdAd4CpDVsvU1iLB6H3UT2JQRnPQIDAQAB
```

**System Info:**

Haraka | Haraka.js — Version: 3.0.0
 --- | :--- 
Node | v16.19.0
OS | Linux dev.port87.com 5.15.0-58-generic #64-Ubuntu SMP Thu Jan 5 11:43:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
openssl | OpenSSL 3.0.2 15 Mar 2022 (Library: OpenSSL 3.0.2 15 Mar 2022)

**Additional context**
The debug statement comes from this line:
https://github.com/haraka/Haraka/blob/master/dkim.js#L307

Here is the DKIM signature from the message:

```
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple;
        d=dev.port87.com; s=oct2021;
        h=from:subject:date:message-id:to:mime-version;
        bh=vWC1aXwKmP/4ijDvWqQqRFK6LiJg2oTbsiNXWfUyNMg=;
        b=jnNv6sU/rjNXZRcJ1tfK1VIMQBM4vkf3dZaTKQeSF4GNE8RnoK3+Z+IqtmMFnPLD0yCGglt6ZX
        B/9KHDTz0pEnMmgLoIizt4bT8vs+7uzml6uAnoi7GJHkr8DppkyznaiPVZ8U+5cITvo8mZkfJwAh
        OJp6r353WXNxqjMzO3v/03+Eppe5KAoVNvBzy3GpHJJoo80wnLJ99T12CQ6ok/4/LZaMQ4OCSMaw
        1Cx545ip6r/EZcNWl6XZxGInDEzJijLI9IFNXnVNajEa9myCUU93TZ07dGX3H+HAMjGiLSjqJXxA
        VDeHzXQ41/q+RY3dffU4ODIl3DAEC1/pPvfQLnSg==
```

--- a/dkim.js
+++ b/dkim.js
@@ -302,7 +302,8 @@ class DKIMObject {
                 }
             }
             if (!res) return this.result('no key for signature', 'invalid');
-            for (const record of res) {
+            for (const recordSegments of res) {
+                const record = recordSegments.join('');
                 if (!record.includes('p=')) {
                     this.debug(`${this.identity}: ignoring TXT record: ${record}`);
                     continue;

